### PR TITLE
refs #11425 - adding lookup_value_matcher to host and hostgroup to make db schema more logical

### PR DESCRIPTION
**Pull request by unorthodoxgeek**

### CODE REVIEW

File: db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class AddLookupValueMatchToHostAndHostgroup < ActiveRecord::Migration
+  def up
+    add_column :hosts, :lookup_value_matcher, :string
+    add_column :hostgroups, :lookup_value_matcher, :string
+
+    Host::Managed.where(:type => 'Host::Managed').find_in_batches(:batch_size => 100) do |group|
+      group.each do |host|
```
Comment (dLobatog) [inline db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb:7]: host instead of group

Comment (unorthodoxgeek) [inline db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb:7]: it's actually group of items, batch would have been a better name

Comment (unorthodoxgeek) [inline db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb:7]: (I was thinking about the Array#in_groups_of method, thus the var name group)

File: db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class AddLookupValueMatchToHostAndHostgroup < ActiveRecord::Migration
+  def up
+    add_column :hosts, :lookup_value_matcher, :string
+    add_column :hostgroups, :lookup_value_matcher, :string
+
+    Host::Managed.where(:type => 'Host::Managed').find_in_batches(:batch_size => 100) do |group|
```
Comment (dLobatog) [inline db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb:6]: `find_each` should suffice if you want to use batches (find_each uses 1000), did you find this migration slow?

Comment (unorthodoxgeek) [inline db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb:6]: not on my local DBs, but it's good practice to use batches in migrations which can iterate over large number of items.

**Comment by dLobatog** [comment]
For the record, this comes from Rails 4. Merging this will make the difference between HEAD:develop & Rails 4 a tad more manageable, thanks :smile: !

@unorthodoxgeek Mind to change the commit to "Fixes" and make the `set_lookup_value_matcher` method private? Should be ready to merge after that. :bomb:

**Comment by unorthodoxgeek** [comment]
@eLobato - done, albeit I made it protected, because in Hostgroup#clone we call that method on the new hostgroup (i.e. not self, but same class, thus, protected)

P.S.
This is the first time I ever use this strange Ruby feature. I was sure it was utterly useless.

**Comment by dLobatog** [comment]
Merged as 4985f94039de8da17e269816cfd4d7788b762132, thanks @unorthodoxgeek!

**Pull request closed without merging**